components/application-broker/internal/syncer/validator.go

```diff
@@ -8,11 +8,74 @@ import (
 	appTypes "github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
 )
 
+type appCRValidatorV2 struct{}
+
+// Validate validates Application custom resource.
+func (v *appCRValidatorV2) Validate(app *appTypes.Application) error {
+	var messages []string
+
+	for _, svc := range app.Spec.Services {
+		if len(svc.Entries) == 0 {
+			messages = append(messages, fmt.Sprintf("Service with id %q is invalid. Entries list cannot be empty", svc.ID))
+			continue
+		}
+
+		APIEntryCnt := 0
+		EventEntryCnt := 0
+		for _, entry := range svc.Entries {
+			switch entry.Type {
+			case apiEntryType:
+				APIEntryCnt++
+
+				if entry.GatewayUrl == "" {
+					messages = append(messages, fmt.Sprintf("Service with id %q is invalid. GatewayUrl field is required for API type", svc.ID))
+				}
+				if entry.TargetUrl == "" {
+					messages = append(messages, fmt.Sprintf("Service with id %q is invalid. TargetUrl field is required for API type", svc.ID))
+				}
+				if entry.Name == "" {
+					messages = append(messages, fmt.Sprintf("Service with id %q is invalid. Name field is required for API type", svc.ID))
+				}
+
+			case eventEntryType:
+				EventEntryCnt++
+			}
+		}
+
+		if APIEntryCnt == 0 && EventEntryCnt == 0 {
+			messages = append(messages, fmt.Sprintf("Service with id %q is invalid. Requires at least one API or Event entry", svc.ID))
+		}
```

Comment:
First, number of Entries is check (`len(svc.Entries) == 0`) then the concrete amount of type is checked, it means we can expect other not supported types so maybe it is good to log (`default case`) other types which can appear in Entries